### ISSUE TITLE
Problem making contract based on dynamic data

### ISSUE THREAD

**Issue by jwvanderbeck** [jrossignol/ContractConfigurator#178]
Hey there,

I'm getting close to finishing up TestFlight support, but i'm running into yet another problem.  I am trying to determine what `part` that the contract will be for dynamically at runtime, but it appears as though CC is only validating my parameters when KSP first loads.

https://github.com/jwvanderbeck/TestFlight/blob/10-contract_configurator/TestFlightContracts/TestFlightContractConfigurator.cs#L146
and
https://github.com/jwvanderbeck/TestFlight/blob/10-contract_configurator/TestFlightContracts/TestFlightContractConfigurator.cs#L158-L237

Basically, at runtime I want to interrogate the TestFlight scenario to determine which part to assign the contract to based on a simple query language I made.  That way I can say pick parts where the user has little or no data, or parts where they have a lot.  

However CC only seems to load the parameters at KSP start, before a game has even been loaded, and well before my scenario exists.

**Comment by jrossignol** [comment]
Load is called only once when Contract Configurator starts up (on the main menu).  Generate is called each time that a new contract is required.  So you should instead save the partQuery string in you Load function, and call SelectPart each time Generate is called.  That should do it.

**Comment by jrossignol** [comment]
On second thought, I checked again, and that already does seem to be what you're doing...

**Comment by jrossignol** [comment]
Hmmm, are you sure it's not working?  You may want to throw in some debugs, as it does look to be correct - it should be calling SelectPart() each time a new contract is generated.  Perhaps the logic in SelectPart() ends up having the same part(s) chosen each time?

**Comment by jwvanderbeck** [comment]
I think the problem is I was trying to validate the data in Load using your ConfigNodeUtils.  I already walked through the execution before posting the issue, and it was definitely breaking because of when it was running.  There might be other issues, but that one fore sure :)

When is Generate run?

**Comment by jrossignol** [comment]
Should be each time a contract of the given type is generated.

**Comment by jwvanderbeck** [comment]
Sorry, I mean when as in what scene(s) is it run and at what time?  I will need to ensure my scenario is loaded for it, and right now I only load my scenario in the Flight and Editor scenes.

Is Generate run when the user enters the contracts office?

**Comment by jrossignol** [comment]
Ah.  It's run whenever KSP attempts to generate a contract.  Which could be in any scene where a game is loaded, so space center, tracking station or flight.

**Comment by jwvanderbeck** [comment]
Oh anytime.  Lovely :)  Ok thanks.  Let me go back and see if I can find a way to do this that isbn't kludgy.

**Comment by jwvanderbeck** [comment]
Closing this since I have it working now.
